mat-process-api.Tests/V1/Controllers/ProcessDataImagesControllerTests.cs

```diff
@@ -175,5 +182,329 @@ public void given_an_exception_thrown_check_that_the_controller_throws_correct_e
+
+            //act
+            var controllerResponse = _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+            var result = controllerResponse as ObjectResult;
+
+            //assert
+            Assert.NotNull(controllerResponse);
+            Assert.NotNull(result);
+            Assert.AreEqual(500, result.StatusCode);
+        }
+
+        [Test]
+        public void given_an_unexpected_error_would_be_thrown_when_GetProcessImage_controller_method_is_called_then_it_returns_a_correct_error_message()
+        {
+            //arrange
+            string expectedErrorMessage = "An error has occured while processing the request - ";
+
+            var randomErrorNumber = _faker.Random.Int(); //triangulating unexpected exceptions
+            string messageRandomizer = _faker.Random.Hash().ToString(); //triangulating error messages
+
+            try // throw random error
+            {
+                ErrorThrowerHelper.GenerateError(randomErrorNumber, messageRandomizer);
+            }
+            catch (Exception ex) //catch the expected error message
+            {
+                expectedErrorMessage += (ex.Message + " " + ex.InnerException);
+            }
+
+            _mockUsecase.Setup(x => x.ExecuteGet(It.IsAny<GetProcessImageRequest>()))
+                .Returns(() => {
+                    ErrorThrowerHelper.GenerateError(randomErrorNumber, messageRandomizer); // throw that same random test error
+                    return new GetProcessImageResponse(null, DateTime.MinValue, null); //dummy return that will never happen due to error being thrown prior to it. It's only needed to keep compiler happy.
+                });
+
+            var request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
+            _mockGetValidator.Setup(x => x.Validate(It.IsAny<GetProcessImageRequest>())).Returns(new FV.ValidationResult()); //validation successful
+
+            //act
+            var controllerResponse = _processImageController.GetProcessImage(request.processType,request.processRef,request.imageId, request.fileExtension);
+            var result = controllerResponse as ObjectResult;
+            var resultContent = result.Value as string; //unwrap the error message from controller response.
+
+            //assert
+            Assert.NotNull(controllerResponse);
+            Assert.NotNull(result);
+            Assert.NotNull(resultContent);
+            Assert.AreEqual(expectedErrorMessage, result.Value); //assert if the error message matches what controller wrapped up
+        }
+
+        [Test]
+        public void given_a_valid_request_when_GetProcessImage_controller_method_is_called_then_it_calls_usecase()
+        {
+            //arrange
+            GetProcessImageRequest request = MatProcessDataHelper.CreateGetProcessImageRequestObject();
```

Comment:
'<1>'

Like before, the contents of the request object don't matter. 
It's a test about calling the usecase if validator permits it.

> This is a nitpicky thing, but it helps a ton, when reading tests. I've seen there were a lot of changes from:
> `var request = new GetProcessImageRequest();`, to:
> `var request = MatProcessDataHelper.CreateGetProcessImageRequestObject();`
> 
> The reason why this line: `var request = new GetProcessImageRequest();` was left as is rather than changed to be generated from helper was that it helps to signify that this test does not depend on what the request object is, and that the only reason the request object is there is to keep the compiler happy so that the test could be run.
> 
> Now that it's changed the helper, it's no longer clear if the contents of the request object are of any significance to the test. Also the line of code is much longer now with a lot more happening, which makes it visually heavier to read in the future.